components/start_trial_modal/index.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+import {Modal, Button} from 'react-bootstrap';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useHistory} from 'react-router-dom';
+
+import {isModalOpen} from 'selectors/views/modals';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {ModalIdentifiers} from 'utils/constants';
+import {closeModal} from 'actions/views/modals';
+import {getLicenseConfig} from 'mattermost-redux/actions/general';
+import {requestTrialLicense} from 'actions/admin_actions';
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+
+import StartTrialModalSvg from './start_trial_modal_svg';
+
+import './start_trial_modal.scss';
+
+enum TrialLoadStatus {
+    NotStarted = 'NOT_STARTED',
+    Started = 'STARTED',
+    Success = 'SUCCESS',
+    Failed = 'FAILED'
+}
+
+function StartTrialModal(): JSX.Element {
+    const [status, setLoadStatus] = useState(TrialLoadStatus.NotStarted);
+    const {formatMessage} = useIntl();
+    const history = useHistory();
+    const dispatch = useDispatch<DispatchFunc>();
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.START_TRIAL_MODAL));
+    const stats = useSelector((state: GlobalState) => state.entities.admin.analytics);
+
+    const requestLicense = async () => {
+        setLoadStatus(TrialLoadStatus.Started);
+        let users = 0;
+        if (stats && (typeof stats.TOTAL_USERS === 'number')) {
+            users = stats.TOTAL_USERS;
+        }
+        const requestedUsers = Math.max(users, 30);
+        const {error} = await dispatch(requestTrialLicense(requestedUsers, true, true, 'license'));
+        if (error) {
+            setLoadStatus(TrialLoadStatus.Failed);
+        }
+
+        setLoadStatus(TrialLoadStatus.Success);
+        await dispatch(getLicenseConfig());
+        dispatch(closeModal(ModalIdentifiers.START_TRIAL_MODAL));
+        history.push('/admin_console/about/license');
+    };
+
+    const btnText = (status: TrialLoadStatus): string => {
+        switch (status) {
+        case TrialLoadStatus.Started:
+            return formatMessage({id: 'start_trial.modal.loading', defaultMessage: 'Loading...'});
+        case TrialLoadStatus.Success:
+            return formatMessage({id: 'start_trial.modal.loaded', defaultMessage: 'Loaded!'});
+        case TrialLoadStatus.Failed:
+            return formatMessage({id: 'start_trial.modal.failed', defaultMessage: 'Failed'});
+        default:
+            return formatMessage({id: 'start_trial.modal_btn.start', defaultMessage: 'Start 30-day trial'});
+        }
+    };
+
+    return (
+        <>
+            {show && (
```

Comment:
2/5 nit: Can we do something like have this component return `JSX.Element | null`, and then at the return statement here, something like

```js
if (!show) {
  return null
}
return (
  <Modal...
);
```

That would let us avoid some layers of nesting in the code.